### ISSUE TITLE
How to close libhv websocket client ?

### ISSUE THREAD

**Issue by anuragvohraec** [ithewei/libhv#602]
Hi ithewei,

I have created a websocketclient connected to a remote WSS server.
However I am unable to close/stop the client from  the client side, even after calling `stop` method of `hv::WebSocketClient`

![image](https://github.com/user-attachments/assets/712ae2b4-37d8-4d4a-a562-b424717900f2)

Because of this my client keeps running even when the code is all executed.

**Comment by ithewei** [comment]
Normally, WebSocketClient::close or stop will disconnect. You can try the [examples/websocket_client_test.cpp](https://github.com/ithewei/libhv/blob/master/examples/websocket_client_test.cpp)

**Comment by anuragvohraec** [comment]
I have put debug points.
![image](https://github.com/user-attachments/assets/e1ea3a99-c612-4205-82e3-cc2b8bf69463)
Only `Here1` gets printed.
`Here2` is never reached!

**Comment by ithewei** [comment]
Stop will join and wait EventLoopThread exit, you can add some print in EventLoopThread::loop_thread function to diagnose why the thread didn't exit.

**Comment by anuragvohraec** [comment]
Ok so I fixed the issue at my end. 
Here is my conclusion based on my understanding after several trials:

When one creates an instance of `new hv::WebSocketClient` without passing any `eventloop` , it creates a default event loop and ends it immediately. But for some reason the socket connection is not released, even when the event loop is already ended. That is why the program was not exiting.

So I created a `hv::EventLoopPtr` inside a seperate thread and made it in running status inside the thread. Gots its pointer via promise and than passed it to `hv::WebSocketClient`, which now do not automatically closes a **running** event loop pointer(which would otherwise if the event loop pointer was not in running status, before passing to its constructor)
![image](https://github.com/user-attachments/assets/f3a47383-affb-4368-b274-c70d41a7c2d4)

Now one can uses it to create the websockets:
` wsc=std::shared_ptr<hv::WebSocketClient>(new hv::WebSocketClient(loop));`
as the loop is already in running state so it won't stop ths loop on its own.

For me the issue is resolved. 
If the libhv is expected to behave this way, than I am happy to close these issue.
